packages/commonwealth/client/scripts/views/pages/view_proposal/index.tsx

```diff
@@ -148,6 +142,62 @@ const ViewProposalPage = ({
     }
   }, [isAdapterLoaded, proposalId]);
 
+  useNecessaryEffect(() => {
+    const afterAdapterLoaded = async () => {
+      try {
+        const proposalFromStore = getProposalFromStore();
+        setProposal(proposalFromStore);
+        setError(null);
+      } catch (e) {
+        console.log(`#${proposalId} Not found in store. `, e);
+      }
+    };
+
+    if (!isAdapterLoaded) {
+      app.chainAdapterReady.on('ready', () => {
+        setIsAdapterLoaded(true);
+        afterAdapterLoaded();
+      });
+    } else {
+      afterAdapterLoaded();
+    }
+  }, [isAdapterLoaded, proposalId]);
+
+  return {
+    error,
+    metadata,
+    isAdapterLoaded,
+    proposal,
+    title,
+    description,
+    isFetchingMetadata,
+  };
+}
+
+type ViewProposalPageAttrs = {
+  identifier: string;
+  type?: string;
+};
+
+const ViewProposalPage = ({
```

Comment:
we have two `ViewProposalPage` components declared in the codebase right now which is misleading. One is here and another in `view_snapshot_proposal/index.tsx`